src/main/java/com/yahoo/sketches/counting/FrequentItems.java

```diff
@@ -0,0 +1,107 @@
+package com.yahoo.sketches.counting;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * This class implements a simple frequent direction algorithm.
+ * It is not intended to be very fast but rather to provide a 
+ * correctness baseline for faster versions 
+ * 
+ * The algorithm is most commonly known as the "Misra-Gries algorithm", 
+ * "frequent items" or "space-saving". It was discovered and rediscovered and redesigned several times over the years.
+ * "Finding repeated elements", Misra, Gries, 1982 
+ * "Frequency estimation of internet packet streams with limited space" Demaine, Lopez-Ortiz, Munro, 2002
+ * "A simple algorithm for finding frequent elements in streams and bags" Karp, Shenker, Papadimitriou, 2003
+ * "Efficient Computation of Frequent and Top-k Elements in Data Streams" Metwally, Agrawal, Abbadi, 2006
+ * 
+ * @author edo
+ */
+public class FrequentItems {
+
+  private PositiveCountersMap counters;
+  private int maxSize;
+  private int maxError;
+  
+  /**
+   * @param maxSize (must be positive)
+   * Gives the maximal number of positive counters the sketch is allowed to keep.
+   * This should be thought of as the limit on its space usage. The size is dynamic.
+   * If fewer than maxSize different keys are inserted the size will be smaller 
+   * that maxSize and the counts will be exact.  
+   */
+  public FrequentItems(int maxSize) {
+    assert(maxSize > 0);
+    this.maxSize = maxSize;
+    counters = new PositiveCountersMap();
+    this.maxError = 0;
+  }
+  
+  /**
+   * @return the number of positive counters in the sketch.
+   */
+  public long nnz() {
```

Comment:
It is bad enough to have a totally unintelligible internal variable name, but unacceptable to expose these names as a public method name.   You have to get out of the writing mathematical-symbols habit, and use descriptive names.